src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php

```diff
@@ -61,6 +64,14 @@
             ])
             ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => DateTimeValueResolver::class])
 
+        ->set('argument_resolver.request_payload', RequestPayloadValueResolver::class)
+            ->args([
+                service('serializer')->nullOnInvalid(),
+                service('serializer')->nullOnInvalid(),
+                service('validator')->nullOnInvalid(),
+            ])
+            ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => RequestPayloadValueResolver::class])
```

Comment:
this must be registered as a pinned value resolver

Reply:
fixed, but still didn't get why value resolvers for `#[MapDateTime]` or `#[CurrentUser]` uses `controller.argument_value_resolver` but my should use `controller.pinned_value_resolver`. What the difference?

Reply:
And this change breaks tests now :crying_cat_face:

Reply:
MapDateTime => because the resolver don't mandate using the attribute to do its magic
CurrentUser => the attribute is not mandatory, one can use `UserInterface` instead

Reply:
Thanx for explanation, tests have been fixed